src/main/java/org/tugraz/sysds/runtime/lineage/LineageItemUtils.java

```diff
@@ -122,33 +128,33 @@ public static Data computeByLineage(LineageItem root) {
 		ProgramBlock pb = new ProgramBlock(new Program());
 		Dag<Lop> dag = new Dag<>();
 		Lop lops = out.constructLops();
-		lops.addToDag( dag );
+		lops.addToDag(dag);
 		pb.setInstructions(dag.getJobs(null,
-			ConfigurationManager.getDMLConfig()));
+				ConfigurationManager.getDMLConfig()));
 		
 		//execute instructions and get result
 		pb.execute(ec);
 		return ec.getVariable(varname);
 	}
 	
-	private static void rConstructHops(LineageItem item, HashMap<Long,Hop> operands) {
-		if( item.isVisited() )
+	private static void rConstructHops(LineageItem item, HashMap<Long, Hop> operands) {
+		if (item.isVisited())
 			return;
 		
 		//recursively process children (ordering by data dependencies)
-		if( !item.isLeaf() )
-			for( LineageItem c : item.getInputs() )
+		if (!item.isLeaf())
+			for (LineageItem c : item.getInputs())
 				rConstructHops(c, operands);
 		
 		//process current lineage item
 		//NOTE: we generate instructions from hops (but without rewrites) to automatically
 		//handle execution types, rmvar instructions, and rewiring of inputs/outputs
-		switch( item.getType() ) {
+		switch (item.getType()) {
```

Comment:
Which functionality should I implement here?
Is it sufficient to raise a RuntimeError if we try to deserialize a lineage trace including dedup items.
Otherwise I have to implement more logic for deserialize dedup items from different files and find the right one at runtime.

Reply:
just add a default branch that throws a controlled exception indicating the non-implemented type.